rust/parquet/src/arrow/schema.rs

```diff
@@ -583,48 +626,109 @@ impl ParquetTypeConverter<'_> {
     }
 
     fn from_int32(&self) -> Result<DataType> {
-        match self.schema.get_basic_info().converted_type() {
-            ConvertedType::NONE => Ok(DataType::Int32),
-            ConvertedType::UINT_8 => Ok(DataType::UInt8),
-            ConvertedType::UINT_16 => Ok(DataType::UInt16),
-            ConvertedType::UINT_32 => Ok(DataType::UInt32),
-            ConvertedType::INT_8 => Ok(DataType::Int8),
-            ConvertedType::INT_16 => Ok(DataType::Int16),
-            ConvertedType::INT_32 => Ok(DataType::Int32),
-            ConvertedType::DATE => Ok(DataType::Date32),
-            ConvertedType::TIME_MILLIS => Ok(DataType::Time32(TimeUnit::Millisecond)),
-            ConvertedType::DECIMAL => Ok(self.to_decimal()),
-            other => Err(ArrowError(format!(
-                "Unable to convert parquet INT32 logical type {}",
-                other
+        match (
+            self.schema.get_basic_info().logical_type(),
```

Comment:
nit: perhaps we can first "merge" the logical and converted type into a logical type and then do the conversion, to avoid some of the code duplications. In the case when logical type is not present, we can always convert the converted type into a logical type while losing some information.

We can do this as a follow-up though.

Reply:
Yeah, I can do it as a follow-up when I've completed the overall 2.6.0 type support